inc/config/dynamic-datasource-trait.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+namespace RemoteDataBlocks\Config;
+
+trait DynamicDatasource {
+	private string $slug;
+	private string $uuid;
+
+	public function set_slug( string $slug ): void {
+		$this->slug = $slug;
+	}
+
+	public function get_slug(): string {
+		return $this->slug;
+	}
```

Comment:
What is the purpose of the slug?

cc @shekharnwagh

Reply:
This was meant to be the predictable unique identifier, so that we can reference the credentials in example code using this, and then users could tag the data source in the Settings Page with this slug. So the slug would act as a link between those 2 systems. Earlier we had `uuid` as unique identifier, but that is not predictable until user adds the data source from Setting Page. Open to other ideas if they seem easier, this was just the first thought that popped into my mind.